### ISSUE TITLE
Telegram Desktop Linux doesn't work with nimf

### ISSUE THREAD

**Issue by yangbeom** [telegramdesktop/tdesktop#2603]
I used UIM-byeoru, Nimf, fcitx, XIM, Ibus. 
but only can type korean to using fcitx.
I wanna using [Nimf](https://github.com/cogniti/nimf). 
so.. Do you have plan to support other input method??

### Configuration
**Operating system:**
Ubuntu 16.04.1 LTS
**Version of Telegram Desktop:**
0.10.19

**Comment by stek29** [comment]
Does Nimf work in other Qt apps?

**Comment by yangbeom** [comment]
may be...
because [Nimf issue #16](https://github.com/cogniti/nimf/issues/16) developers say you need patch like support fcitx.
so i tried it but fail :(
[Nimf developer's patch](https://github.com/cogniti/nimf/files/390379/qtbase_5_6_0-with-nimf.txt)

**Comment by stek29** [comment]
So TDesktop would need one more patch to Qt which must be rewritten because TDesktop already patches Qt and patches are not compatible, and all that just to support a new input method no one uses?

**Comment by yangbeom** [comment]
i tried test Nimf, But Build fail Nimf.
Nimf developers fix it in jan/2017
i will reply comment when i build Nimf and finish some test

**Comment by mxa** [comment]
Is this fixed? I still can't use Nimf in Telegram Desktop.

**Comment by yangbeom** [comment]
maybe not.... I forgot it.... didn't test yet.

**Comment by yangbeom** [comment]
I tried to write korean using Nimf(2017.04.08) in Telegram Desktop, But Doesn't work.

**Comment by stek29** [comment]
Then why won't you reopen the issue?

**Comment by mxa** [comment]
@yangbeom should reopen this issue.

**Comment by yangbeom** [comment]
@mxa Sure :)

**Comment by mxa** [comment]
:) maybe change the title in "Telegram Desktop Linux doesn't work with nimf"

**Comment by mxa** [comment]
Maybe @auchri can label this one bug, help-wanted, linux too?

**Comment by auchri** [comment]
@mxa maybe 😉

**Comment by mxa** [comment]
@stek29 "Does Nimf work in other Qt apps?" I tried Qtractor, and yes, I can name tracks there using Nimf.

**Comment by devunt** [comment]
Nimf works with Kate, so it works in other Qt apps.

log from `G_MESSAGES_DEBUG=nimf kate`:
```
(kate:3317): nimf-DEBUG: im-nimf-qt5.cpp:463: NimfInputContextPlugin::NimfInputContextPlugin()
(kate:3317): nimf-DEBUG: im-nimf-qt5.cpp:481: virtual QPlatformInputContext* NimfInputContextPlugin::create(const QString&, const QStringList&)
(kate:3317): nimf-DEBUG: im-nimf-qt5.cpp:251: NimfInputContext::NimfInputContext()
(kate:3317): nimf-DEBUG: nimf-im.c:298: nimf_im_new
```

whereas running `G_MESSAGES_DEBUG=nimf telegram-desktop` shows nothing.

**Comment by devunt** [comment]
`strace kate` includes:
```
stat("/usr/lib/qt/plugins/platforminputcontexts/.", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
open("/usr/lib/qt/plugins/platforminputcontexts", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = 7
...
lstat("/usr/lib/qt/plugins/platforminputcontexts/libqt5im-nimf.so", {st_mode=S_IFREG|0644, st_size=39848, ...}) = 0
open("/usr/lib/qt/plugins/platforminputcontexts/libqt5im-nimf.so", O_RDONLY|O_CLOEXEC) = 7
```

but `strace telegram-desktop` does not have such entries.

**Comment by stek29** [comment]
@devunt because TDesktop uses his own static patched Qt

**Comment by mxa** [comment]
@devunt is there a way to make nimf and the compose key work? Both are there because of that static patched Qt https://github.com/telegramdesktop/tdesktop/issues/2193

**Comment by mxa** [comment]
B U M P

**Comment by mxa** [comment]
The Qt input modules for nimf can be found [here](https://github.com/cogniti/nimf/tree/master/modules/clients)

**Comment by BasixKOR** [comment]
This issue happens in ibus too. I think this issue is not just limited to a specific input method.

**Comment by 5HARK** [comment]
Check this documents
https://gist.github.com/5HARK/2d33ed72bdfd488dc20c51ec11dd209b

**Comment by 5HARK** [comment]
Nimf repository is moved.
https://gitlab.com/nimf-i18n/nimf

**Comment by 5HARK** [comment]
I wrote PR for this issue.

**Comment by 5HARK** [comment]
This issue has fixed on [Telegram Desktop 1.5++](https://github.com/telegramdesktop/tdesktop/releases/tag/v1.5.0).
